### DISCUSSION TITLE
Address or Undefined Sanitizer with Clang

### DISCUSSION THREAD

**Discussion by NikLeberg** [nickg/nvc#1054] [Q&A]
I tried to compile NVC with `clang` and enable the flags `--enable-asan` or `--enable-ubsan` which produced an ungodly amount of runtime errors detected by either sanitizers.

If I compile NVC with `gcc` and only enable `--enable-asan`, then it seems to work ok.

What combinations are supported? Or is clang not supported at all?

**Comment by NikLeberg** [comment]
I see that for the CI leak detection is turned off:
https://github.com/nickg/nvc/blob/5b805daac26765af7adac7d8cac6b2b512212cdd/.github/workflows/build-test.yml#L36

Is there a specific reason? Are all detected leaks false-positives? Or should one have a closer look?

**Comment by nickg** [comment]
> What combinations are supported? Or is clang not supported at all?

Clang checks some different things and I normally build with GCC. I think you mean ubsan errors here? Asan shouldn't have errors with either compiler. I fixed most of the ubsan warnings from Clang on my Mac. The one about unaligned access in `interp_store` is harmless.

> Is there a specific reason? Are all detected leaks false-positives? Or should one have a closer look?

They are mostly benign leaks of objects that are allocated once and then persist until exit. I check these occasionally but I don't think there's anything worth worrying about.
